Add/url input variation

TheSteveK opened the pull request:
Building off the email block, this PR adds another variation of the text input form for a URL input.

To add a new block, add a variation to /block-editor/text-input/variations.js  Add a validator to blocks/text-input/validations.js (if required) and then add an icon to /icons

You then end up with something like this:

<img width="679" alt="Screen Shot 2022-03-25 at 10 54 37 AM" src="https://user-images.githubusercontent.com/12895386/160147983-c7817839-0e38-4637-ab0c-4cb096cfdf63.png">

Testing instructions:
Apply this PR and do the yarn thing then visit https://crowdsignal.localhost:9000/project and create a new project.

Click the block inserter and you should see the new URL form input block, add it and then preview your form.

Try adding an invalid URL and see if the error message appears

Correct the URL, verify the error goes away

Note, right now the validator doesn't look for http(s):// but if anyone has a strong feeling about that, the regex can be updated to require it.

digitalwaveride reviewed (commented):
Nice Steve! Works for me! 👍 

If the code review is fine ... 🚀

CGastrell commented on packages/block-editor/src/text-input/variations.js:
```diff
@@ -31,4 +31,27 @@ export default [
+		),
+		icon: <URLInputIcon />,
+		attributes: {
+			label: __( 'Enter url', 'block-editor' ),
```
Should we use "URL" instead of "url"?

CGastrell commented on packages/blocks/src/text-input/validations.js:
```diff
@@ -8,4 +8,14 @@ export default {
+				/[(http(s)?):\/\/(www\.)?a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)/gi
+			)
+		) {
+			return 'Please enter a valid url.';
```
Needs i18n.

I'd also repeat the question regarding the use of all caps on acronyms, should it be "URL"?

CGastrell commented on packages/blocks/src/text-input/validations.js:
```diff
@@ -8,4 +8,14 @@ export default {
 			return 'Please enter a valid email address.';
```
Not part of this PR, but I just noticed this one doesn't have a translation either

CGastrell reviewed (changes requested):
Minor translation and question, but it looks good. Nice job!

TheSteveK commented:
I addressed @CGastrell 's notes (including URL in text).  Technically it should've been another PR, but I went ahead and added i18n to the email error message as well as the URL one.

CGastrell commented on packages/blocks/src/text-input/validations.js:
```diff
@@ -1,11 +1,26 @@
 			)
 		) {
-			return 'Please enter a valid email address.';
+			return __( 'Please enter a valid email address.' );
```
Translation calls should also carry a domain (second argument). I think we're matching domains to the package names, so `blocks` should the domain here.

CGastrell commented on packages/blocks/src/text-input/validations.js:
```diff
@@ -1,11 +1,26 @@
+				/[(http(s)?):\/\/(www\.)?a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)/gi
+			)
+		) {
+			return __( 'Please enter a valid URL.' );
```
Same as above, translation call lacks domain (sorry I didn't specify this on my last review)

CGastrell reviewed (changes requested):
Works nicely, I just pointed out the lack of domain parameter on the translation calls.
